### DISCUSSION TITLE
v4.2.5

### DISCUSSION THREAD

**Discussion by osy** [utmapp/UTM#5244] [Releases]
## Highlights
* (iOS SE) **TCTI backend updates**. Vector instruction support can improve the performance of certain workloads.
* (macOS) **Guest interface commands**. New commands accessible from the scripting interface as well as the command line interface (utmctl) can be used to read/write files, execute commands, and list IP addresses. These commands require QEMU guest agent to be installed.
* (macOS) **Scripting interface additions**. The scripting interface now allows for creation of a new VM, configuration an existing VM, changing drive image, and more. See [this page](https://docs.getutm.app/scripting/reference/) for more details and [the cheat sheet](https://docs.getutm.app/scripting/cheat-sheet/) for example usage.
* (macOS) **External read-write drive images**. In QEMU, if a supported drive interface is selected (USB, Floppy, etc) along with an image type of Disk, the drive can now be marked as either read-only or read-write. This can be used as an alternative way of transferring data to and from the VM when the guest does not support SPICE or VirtFS. In AVF, support for read-write images has be fixed as well (when "read only" is unchecked in the settings).

## Changes (v4.2.5)
* Reverted ANGLE to an older version which includes the fix detailed in v4.2.4 but also does not break the Metal backend (#5238)
* Translation: updated Polish (thanks @matgaj, @mavethee)
* (macOS) Fixed an issue where the pointer device stopped working on macOS Monterey guests (#5237)
* (macOS) AVM: Added option to use emulated trackpad (requires macOS Ventura guest)

## Changes (v4.2.4)
* UTM will now display the release notes the first time a new version is launched
* Updated ANGLE which should fix some GPU acceleration issues on Linux (please see known issues below for unresolved issues)
* Fixed an issue where a removable drive or VirtIO shared directory was set up when QEMUHelper was already running and therefore was not able to create a bookmark and an "access denied" error shows up on startup
* Introduced a new preference setting for using the Option key as Meta in the built in console. This was on by default in previous versions and is not off by default which should allow international text to be entered properly (#5202)
* Wizard: Make virtio-gpu-pci default for Linux (again) which was reverted a few releases ago in order to address some GPU issues
* Translation: updated Polish (thanks @matgaj)
* (iOS) Fixed built in terminal overlapping in the safe area

## Changes (v4.2.3)
* Added support for read-write removable devices on supported interfaces (USB, Floppy, etc). To use this feature, set the image type to "Disk" and uncheck the "Read Only?" option in the drive settings. Note that you may have to eject and re-mount the drive image for effect to take place. (#5079)
* Removable USB devices will no longer show up as a CD drive if the image type is set to "Disk" (#5171)
* Reverted a change from v4.2.0 where virtio bus devices were removed (this broke some people's configurations) (#5137)
* Fixed an issue where when QEMU fails to start, subsequent attempts to start the VM will automatically fail
* Fixed an issue where due to a race, removable drives may not be mounted on start
* (iOS) Fixed an issue when the pointer is hovering over a toolbar button when the VM starts and subsequent clicks will toggle the button (#4843)
* (macOS) AVF: Fixed an issue where a removable drive that is not marked as read-only still gets marked as read-only after restarting UTM (when upgrading from a previous version, you must eject and re-mount the image for effect to take place) (#5170)
* (macOS) Fixed an issue where incorrect subnet parameters are generated for vmnet advanced settings (#5128)
* (macOS) CLI: Added a new `clone` command (#5004)

## Changes (v4.2.2)
* Fixed an issue where a VM that is moved manually (e.g. with Finder) cannot be started (#5121)
* Fixed an issue where the cursor disappears after resuming a suspended VM (#4665)
* (macOS) Scripting: Add support for `make`, `duplicate`, and `delete` verbs on virtual machines
* (macOS) Scripting: Add support for modifying certain configuration options
* (macOS) Scripting: Removed virtual machine properties: `notes`, `machine`, `architecture`, `memory`. You should now use similar properties in the `configuration` record
* (macOS) Support changing the audio backend to CoreAudio and make CoreAudio the default (due to better latency and as a workaround to some crashes) (#5125)

## Changes (v4.2.1)
* QEMU: Support more than 8 cores for ARM64 targets (#4847)
* Synchronize time on resume when QEMU guest agent is installed which should fix clock drift issues (#3218)
* (iOS SE) Updated TCTI to latest branch from @ktemkin which has support for vector instructions (#4865)
* (macOS) Switch to emulated trackpad when guest+host are macOS 13+ (second attempt for #4636 and #4645)
* (macOS) Fixed the "raw image" checkbox when creating a new drive (#5100)
* (macOS) Add PC speaker sound device as an option. You can now add a second sound device on supported targets (x86_64/i386 PC based targets) and set the hardware to PC Speaker to emulate older motherboard speakers.
* (macOS) Scripting: new verbs "open file", "execute", and "query ip" which uses the QEMU guest agent to send commands to the guest. Use the Apple Script Editor's Open Dictionary tool for more information.
* (macOS) CLI: new commands "file push", "file pull", "exec", and "ip-address". See `utmctl help` for more information.

## Changes (v4.2.0)
* Updated virglrenderer to latest (27/02/2023) commit
* Automatically revert CPU to "default" if model is missing due to updated QEMU (#4910)
* ~~Remove virtio-bus devices (they didn't work) (#4893)~~
* Support drag and drop to import an .utm package (#3599)
* Support copying sparse images across different volumes (thanks @ktprograms)
* Fixed tearing/performance issue for non-GL accelerated displays when under graphics load (#5049)
* Fixed error when importing a new drive with the same name as an existing drive (#5052)
* ~~Fixed a crash on startup when multiple audio devices are plugged in (#4629)~~
* Wizard: Generate a working configuration for isapc (#4961)
* Settings: Use pretty labels for CPU flags (#4940)
* Translation: updated French (thanks @FRtranslator)
* Translation: added Czech readme (thanks @stinovlasmp)
* Translation: updated German (thanks @LostInCompilation and @afh)
* (macOS) Capture Mouse has been renamed to Capture Input and now will show the current capture status as a toggle button
* (macOS) Apple VMs (macOS guests) now support Capture Input which will capture the system keys. This used to be always on as the default. It is now off by default and the toggle button will enable it. (#4667)
* (macOS) Ignore missing links (drives, shared directory, etc) when saving an Apple VM
* (macOS) Preference: added option to force Num Lock on (#4625)
* (macOS) Preference: added option to prevent idle sleep while VM is running (#4963)
* (macOS) Preference: added option to suppress exit confirmation (#4989)
* (macOS) Fixed delete drive confirmation not showing up on latest macOS
* (macOS) Fixed disposable mode stuck on when a VM is run once in disposable mode (#5025)
* (macOS) Fixed an issue where a moved or deleted VM with a stopped window open will attempt to use the old path (#4985)
* (macOS) Fixed cursor scrolling and double click issues (#4636, #4645)
* (macOS) Support following a change in system default input or output audio device (#3488)
* (macOS) Support retry of macOS installation when it fails (the IPSW will not automatically be unlinked)
* (macOS) Support retaining of installer IPSW bookmark across VM moves and UTM restarts (#4938)

## Issues
Please check the full list on [Github](https://github.com/utmapp/UTM/issues) and help report any bug you find that is not listed.

* There is a known bug in newer versions of Mesa which prevents some Wayland applications (such as Firefox) from launching when GPU acceleration is enabled. One workaround is to switch to Xorg from the log-in screen (typically by clicking on the gear icon) and another is to patch Mesa to a newer version.
* Debian 12 fails to load the GUI installer when GPU acceleration is enabled. Please use the text installer instead.

## Installation

* [iOS](https://docs.getutm.app/installation/ios/)
* [macOS](https://docs.getutm.app/installation/macos/)

| File | Description | Installation | JIT | Hypervisor | USB |
|------|------------|--------------|-----|-----------|-----|
| UTM.deb | Jailbroken iOS version | Open in Cydia, dpkg, or Sileo | Yes | Yes(1) | Yes |
| UTM.dmg | macOS version | Mounting and copying UTM.app to /Applications | Yes | Yes | Yes |
| UTM.ipa | Non-jailbroken iOS version (sideloading) | AltStore, etc (see guide) | Yes(2) | No | No |
| UTM.HV.ipa | Non-jailbroken iOS version (TrollStore) | TrollStore | Yes | Yes(1) | Yes |
| UTM.SE.ipa | Non-jailbroken iOS version (sideloading) | AltStore, enterprise signing, etc | No | No | No |

1. Hypervisor on iOS requires an M1 iPad.
2. Enabling JIT may require a separate JIT enabler such as [Jitterbug][2] or Jitstreamer.

[1]: https://getutm.app/install/
[2]: https://github.com/osy/Jitterbug

<hr /><em>This discussion was created from the release <a href='https://github.com/utmapp/UTM/releases/tag/v4.2.5'>v4.2.5</a>.</em>

**Comment by macuserguru** [comment]
My first impression

UTM v4.2.5 on M1 MacMini, Ventura 13.3.1

VM: Debian GNU/Linux 11 (bullseye) aarch64 
Spice client folder works fine!
Copy & Paste works in both direction.

VM: Fedora Linux 38 (Workstation Edition) aarch64
Some SPICE guest tools installed
Spice client folder don't work
Copy & Paste works in both direction.

Windows ARM
DavWWWRoot .... Spice client folder works fine!
Copy & Paste works in both direction.

All this VMs created under UTM 4.1.x

UTM 4.1.6 are installed from Apples App Store too.

**Comment by tmmvn** [comment]
I see the latest release on the App Store now hits the GitHub API due to the [addition of parsing](https://github.com/utmapp/UTM/commit/db58b588fe42046f22690046f6c0352c66be2ecb) (Firewall prompted for the connection). As the App claims not to collect any data, I am not sure GitHub API is that innocent. Pretty much everything Microsoft does involves collecting data. Has this been confirmed that hitting the GitHub API doesn't leak e.g. user's IP address info and such?

**Reply by osy** [reply]
@tmmvn I’m not sure what GitHub does but note this request is only made once per new release. I can add in a toggle for “do not show again” which will permanently disable it (however it would still make the request on the very first launch). Do you have any better ideas? I think there is value in showing the update change log as most people have auto updates enabled and would not know if an update introduced any changes.

**Reply by tmmvn** [reply]
@osy Well, App Store has quite prominent "Version history" and "What's New" sections for each App and even automatic updates show up in the "Updates" section. So the information is not hard to come by.

I'd probably change it to a generic popup that would mention the app has been updated to a new version. Then give a button to download and show notes (and accepting terms information may be transmitted to third party) or direct where to find the version notes. Or something along those lines. 

Not sure if you are able to differentiate things between App Store and non App Store versions as I guess that would not be quite as neat of a UX. I was also unable to find info on what GitHub might actually record from the incoming request, and if they then use e.g. IP addresses for further analytics / merging from other Microsoft products.

I simply blocked the outgoing request and I'm all set, but simply thought I'd point it out as someone might get crankier. So not sure if it would require updating the "No Data Collected" part on the App Store page to just leave functionality as is. That makes requests less of a surprise.

Many apps that use some analytics or a/b frameworks just say there's "Usage data" and "Diagnostics" "not linked" data collection and call it a day. But from the requests on those apps, seems it might be more identifiable than the devs think. But as said, less of a surprise when the App tries to hit some random location.

**Reply by osy** [reply]
Thanks for the suggestion, I will keep this in mind.

**Reply by LostInCompilation** [reply]
IMO having to accept terms every time just to view the release log is terrible usability. I don't see an issue with analytic data and GitHub API. It's just how it is today, if you want to keep this data secret hide your IP or block the request. But a note that the GitHub API is used would be enough for 99.9% of people.

>I'd probably change it to a generic popup that would mention the app has been updated to a new version. Then give a button to download and show notes (and accepting terms information may be transmitted to third party) or direct where to find the version notes. Or something along those lines.

At some point you just have to decide between functionality and absolute privacy. Showing a release log is definitely preferred by most user compared to not giving away your IP. The IP address is pretty much public data anyway today.
If you really want to protect yourself from such data collection, you should take care of that by hiding your IP and blocking connections. But putting burden upon every user, who simply doesn't care about IP tracking is not a good way to go. Pure privacy is pretty much an illusion as soon as you use internet services and Apps which communicate with them.

**Reply by osy** [reply]
Yeah, I think the privacy policy and app privacy transparency would have to be updated. I need to get clarification from GitHub support regarding exactly what PII is collected.

**Reply by osy** [reply]
@tmmvn The privacy policy has been updated: https://mac.getutm.app/privacy/ with details from GitHub. Additionally, we've requested an update from Apple to the App Privacy section of the App Store page.
